pkg/reconciler/revision/resources/imagecache.go

```diff
@@ -28,7 +28,7 @@ import (
 
 // MakeImageCache makes an caching.Image resources from a revision.
 func MakeImageCache(rev *v1.Revision) *caching.Image {
-	image := rev.Status.ImageDigest
+	image := rev.Status.DeprecatedImageDigest
```

Comment:
We should be producing a `caching.Image` for each container image

Reply:
True  we need to handle caching for each container
I have not done those changes as part of this PR 
Will send the image cache part after this PR merge

Reply:
sounds good!